src/aspire/image/xform.py

```diff
@@ -319,6 +321,67 @@ def _forward(self, im, indices):
 
         return im
 
+    @abc.abstractproperty
+    def noise_var(self):
+        """
+        Concrete implementations are expected to provide a method that
+        returns the noise variance for the NoiseAdder.
+        """
+
+
+class CustomNoiseAdder(NoiseAdder):
+    """
+    Instantiates a NoiseAdder using the provided `noise_filter`.
+    """
+
+    @property
+    def noise_var(self):
+        """
+        Return noise variance.
+
+        CustomNoiseAdder will estimate noise_var by taking a sample of the noise.
+
+        If you require tuning the noise_var sampling, see `get_noise_var`.
+        """
+        return self.get_noise_var()
+
+    def get_noise_var(self, sample_n=100, sample_res=128):
+        """
+        Return noise variance.
+
+        CustomNoiseAdder will estimate noise_var by taking a sample of the noise.
+
+        :sample_n: Number of images to sample.
+        :sample_res: Resolution of sample (noise) images.
+        """
+        im_zeros = Image(np.zeros((sample_n, sample_res, sample_res)))
+        im_noise_sample = self._forward(im_zeros, range(sample_n))
+        return np.var(im_noise_sample.asnumpy())
+
+
+class WhiteNoiseAdder(NoiseAdder):
+    """
+    A Xform that adds white noise, optionally passed through a Filter object, to all incoming images.
+    """
+
+    # TODO, check if we can change seed and/or why not.
+    def __init__(self, var, seed=0):
+        """
+        Return a CustomNoiseAdder instance from `noise_var` and using `seed`.
+
+        :param noise_var: Target noise variance.
+        :param seed: Optinally provide a random seed used to generate white noise.
+        """
+        self._noise_var = var
+        super().__init__(noise_filter=ScalarFilter(dim=2, value=var), seed=seed)
+
+    @property
+    def noise_var(self):
+        """
+        Returns noise variance. Note in this case noise variance is known.
```

Comment:
Sort of a confusing docstring.. This is because the `WhiteNoiseAdder` has been instantiated with a variance value, right? Could we just spell that out?

Reply:
Sure.